README.md

```diff
@@ -1,358 +1,302 @@
+doubleOrZero(leftValue); // => 0 - also logs "The error was Division by Zero!"
+doubleOrZero(rightValue); // => 20
 
-#### <a name="build-taskeither"></a>Building a TaskEither
-
-##### From a value
-
-You can build a TaskEither as you would build an Either: `left`, `right`, `fromNullable`, `fromPredicate`.
-
-```typescript
-  const leftValue = taskEither.left("value");
-  const rightValue = taskEither.right("value");
-  ...
+// Pro-tip: either.match is short for the following:
+const doubleOrZeroBis = flow(
+  either.map((n: number) => 2 * n),
+  either.getOrElse((error: string) => {
+    console.log(`The error was ${error}`);
+    return 0;
+  });
+);
 ```
 
-You can also build a TaskEither from a Promise! You have the `tryCatch` helper, that takes two functions as arguments. The first one returns a promise, and the second one returns the value to put in left if the promise rejects.
+## <a name="taskeither"></a>TaskEither
 
-```typescript
-const asyncIsEven = async (a: number) => {
-  await asyncExternalCall();
+### <a name="taskeither-builders"></a>Create a `TaskEither<E, A>`
 
-  if (a % 2 !== 0) {
-    throw new Error("ODD");
-  }
-  return a;
-};
-const buildTaskEither = (number: number) =>
-  taskEither.tryCatch(
-    () => asyncIsEven(number),
-    () => `${number} is odd`
-  );
+```ts
+// Build an Either
+const leftValue = taskEither.left('value');
+const rightValue = taskEither.right('value');
 
-const rightValue = buildTaskEither(4); // taskEither.right(4)
-const leftValue = buildTaskEither(3); // taskEither.left('3 is odd')
-```
+// Build from a value
+taskEither.fromNullable('value was nullish')(null); // => taskEither.left('value was nullish')
+taskEither.fromNullable('value was nullish')('value'); // => taskEither.right('value')
 
-#### <a name="get-teither"></a>Get Value from a TaskEither
+// Build from a predicate
+const isEven = (num: number) => num % 2 === 0;
 
-In fp-ts, if you invoke a `TaskEither`, you get a `Promise<Either>` so most of the time, what we do is we invoke our TaskEither and then use the aforementioned methods on the `Either` to retrieve its value.
+const taskEitherBuilder = taskEither.fromPredicate(
+  isEven,
+  number => `${number} is an odd number`
+);
 
-```typescript
-const ten = taskEither.right(10); // this is a right branch of a taskEither
-const rightValue = await ten(); // we invoke ten, making it a Promise<Either> and then await the promise, so rightValut is an Either.right
+taskEitherBuilder(3); // => taskEither.left('3 is an odd number')
+taskEitherBuilder(4); // => taskEither.right(4)
 
-const returnedValue = either.getOrElse(() => 0)(rightValue); // 10
-```
+// Convert from Option
+const noneValue = option.none;
+const someValue = option.some('value');
 
-### <a name="arrays"></a>Array and ReadonlyArray
+taskEither.fromOption('value was nullish')(noneValue); // => taskEither.left('value was nullish')
+taskEither.fromOption('value was nullish')(someValue); // => taskEither.right('value')
```

Comment:
I think I would not rewrite all these here and it's 'just' a port of the `Either`. Maybe just a sentence listing that `fromNullable`, `fromPredicate` and `fromOption` exists just like for `Either` could be enough